kernel/src/main/java/org/kframework/parser/json/JsonParser.java

```diff
@@ -296,7 +298,27 @@ public static Att toAtt(JsonObject data) {
         JsonObject attMap = data.getJsonObject("att");
         Att newAtt = Att.empty();
         for (String key: attMap.keySet()) {
-            newAtt = newAtt.add(key, attMap.getString(key));
+            if (key.equals(Location.class.getName())) {
+                JsonArray locarr = attMap.getJsonArray(Location.class.getName());
+                newAtt = newAtt.add(Location.class, Location(locarr.getInt(0), locarr.getInt(1), locarr.getInt(2), locarr.getInt(3)));
+            } else if (key.equals(Source.class.getName())) {
+                newAtt = newAtt.add(Source.class, Source.apply(attMap.getString(key)));
+            } else if (key.equals(Production.class.getName())) {
+                newAtt = newAtt.add(Production.class, (Production) toSentence(attMap.getJsonObject(key)));
+            } else if (key.equals(Sort.class.getName())) {
+                newAtt = newAtt.add(Sort.class, toSort(attMap.getJsonObject(key)));
+            } else if (key.equals("bracketLabel")) {
+                newAtt = newAtt.add("bracketLabel", KLabel.class, toKLabel(attMap.getJsonObject(key)));
+            } else if (key.equals(Att.PREDICATE())) {
+                newAtt = newAtt.add(Att.PREDICATE(), Sort.class, toSort(attMap.getJsonObject(key)));
+            } else if (key.equals("cellOptAbsent")) {
+                newAtt = newAtt.add("cellOptAbsent", Sort.class, toSort(attMap.getJsonObject(key)));
+            } else if (key.equals("cellFragment")) {
+                newAtt = newAtt.add("cellFragment", Sort.class, toSort(attMap.getJsonObject(key)));
+            } else if (key.equals("sortParams")) {
+                newAtt = newAtt.add("sortParams", Sort.class, toSort(attMap.getJsonObject(key)));
+            } else
+                newAtt = newAtt.add(key, attMap.getString(key));
```

Comment:
What's the change that's meant we need to special-case these now? It might be worth pulling this if -else chain out into its own method.

Reply:
I think this location is fine, but you are correct in noticing this isn't pretty.
Some of our attributes can take entire classes: [here](https://github.com/runtimeverification/k/blob/master/kore/src/main/scala/org/kframework/attributes/Att.scala#L65-L66)
which makes the java code very convenient.
Since now we want to serialize and deserialize the entire definition we ran into the issue that we couldn't deserialize some attributes back into the structure they came from. This code special cases those classes (the ones I could find).